criu/unittest/unit.c

```diff
@@ -0,0 +1,74 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+
+
+#include "log.h"
+#include "criu-log.h"
+
+int parse_statement(int i, char *line, char **configuration);
+
+int main(int argc, char *argv[], char *envp[])
+{
+	char **configuration;
+	int i;
+
+	configuration = malloc(10 * sizeof(char *));
+	log_init(NULL);
+
+	i = parse_statement(0, "test test\n", configuration);
+
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--test"));
+	assert(!strcmp(configuration[1], "test"));
+
+	i = parse_statement(0, "dsfa \"aaaaa \\\"bbbbbb\\\"\"\n", configuration);
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--dsfa"));
+	assert(!strcmp(configuration[1], "aaaaa \"bbbbbb\""));
+
+
+	i = parse_statement(0, "verbosity 4\n", configuration);
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--verbosity"));
+	assert(!strcmp(configuration[1], "4"));
+
+	i = parse_statement(0, "lsm-profile \"\" # more comments\n", configuration);
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--lsm-profile"));
+	assert(!strcmp(configuration[1], ""));
+
+	i = parse_statement(0, "lsm-profile \"something\"# comment\n", configuration);
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--lsm-profile"));
+	assert(!strcmp(configuration[1], "something"));
+
+	i = parse_statement(0, "#\n", configuration);
+	assert(i == 0);
+
+
+	i = parse_statement(0, "lsm-profile \"selinux:something\\\"with\\\"quotes\"\n", configuration);
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--lsm-profile"));
+	assert(!strcmp(configuration[1], "selinux:something\"with\"quotes"));
+
+	i = parse_statement(0, "work-dir \"/tmp with spaces\"\n", configuration);
+	assert(i == 2);
+	assert(!strcmp(configuration[0], "--work-dir"));
+	assert(!strcmp(configuration[1], "/tmp with spaces"));
+
+	i = parse_statement(0, "a b c d e f g h i\n", configuration);
+	assert(i == 9);
+	assert(!strcmp(configuration[0], "--a"));
+	assert(!strcmp(configuration[1], "b"));
```

Comment:
I still don't understand an use-case for this. Do we have arguments that can accept multiple values?

I mean --arg-list val1 val2 val3 val3. If we are not, we can simplify the code a lot. We need to get the first word and it will be an argument and other part of the line will be a value. We will not need to handle quotas...

Reply:
You are right. I blindly copied your example, but we do not need to handle it. This can be simplified.

Reply:
The current behaviour of the parser is documented at the end of the man page. There is says to use quotes if the value behind the option contains whitespaces. So we probably want to keep that functionality. Also we support comments after the value to an option `lsm-profile selinux:something #comment`. So we need to scan for whitespace to see if there is a comment at the end. Not sure if we still can simplify the code and keep the existing functionality.